File: app/src/main/java/com/recursia/popularmovies/domain/MoviesListInteractorImpl.java
package com.recursia.popularmovies.domain;

import com.recursia.popularmovies.domain.models.Movie;

import java.util.List;

import io.reactivex.Single;

public class MoviesListInteractorImpl implements MoviesListInteractor {

    private final MoviesRepository moviesRepository;

    public MoviesListInteractorImpl(MoviesRepository moviesRepository) {
        this.moviesRepository = moviesRepository;
    }

    @Override
    public Single<List<Movie>> discoverMovies(String sortBy, int page, int voteCount, String language) {
        return moviesRepository.discoverMovies(sortBy, page, voteCount, language);
    }

}

File: app/src/main/java/com/recursia/popularmovies/domain/MoviesListInteractorImpl.kt
package com.recursia.popularmovies.domain

import com.recursia.popularmovies.domain.models.Movie

import io.reactivex.Single

class MoviesListInteractorImpl(private val moviesRepository: MoviesRepository) : MoviesListInteractor {

    override fun discoverMovies(sortBy: String, page: Int, voteCount: Int, language: String): Single<List<Movie>> {
        return moviesRepository.discoverMovies(sortBy, page, voteCount, language)
    }
}
